src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -159,6 +159,10 @@ private function getMethodsToAutowire(\ReflectionClass $reflectionClass, array $
                     continue 2;
                 }
             }
+
+            if ($reflectionMethod->isAbstract() && !$reflectionMethod->getNumberOfParameters()) {
```

Comment:
Shouldn't we check all conditions making it eligible to autowiring (return type...).